client/src/jobs/job.py

```diff
@@ -66,6 +56,23 @@ class ImageOptions(BaseModel):
         protected_namespaces=(),
     )
 
+    def __init__(self, **data: Any) -> None:
+        spec = data.get("spec")
+        if spec is not None and not Path(spec).is_absolute():
+            if job_path := inspect.stack()[1].filename:
```

Comment:
This makes me a bit uneasy, since it involves the call stack. What if the user extracts their image options in to a function (e.g., so they can be reused across multiple jobs)?

In any case, the idea warrants a comment in the code, IMO.